internal/server/data/migrations.go

```diff
@@ -1329,3 +1330,58 @@ func addAccessKeyIssuedForKind() *migrator.Migration {
 		},
 	}
 }
+
+func resourceURNToDestinationNameAndResource() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2023-01-23T17:45",
+		Migrate: func(tx migrator.DB) error {
+			if migrator.HasColumn(tx, "grants", "resource") {
+				_, err := tx.Exec(`
+					DROP INDEX IF EXISTS idx_grants_subject_privilege_resource;
+					ALTER TABLE grants
+						ADD COLUMN IF NOT EXISTS destination_name text,
+						ADD COLUMN IF NOT EXISTS destination_resource text;`)
+				if err != nil {
+					return err
+				}
+
+				type grantResource struct {
+					ID          uid.ID
+					ResourceURN string
+				}
+
+				rows, err := tx.Query(`SELECT id, resource FROM grants WHERE deleted_at is null`)
```

Comment:
Should we still select the deleted records? That way we can still track what they were after the `resource` column is dropped. Not sure if this could mess with the `deleted_at` timestamp though.

Reply:
I was thinking about that myself. Omitting deleted grants significantly reduces how many grants are migrated but that might not be a good thing in this case. Other migrations touch all records, deleted and not deleted so I think that's what I'll do here too